subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java

```diff
@@ -121,4 +125,20 @@ public static File getClasspathForResource(URL resource, String name) {
         }
         throw new GradleException(String.format("Cannot determine classpath for resource '%s' from location '%s'.", name, location));
     }
+
+    private static URI toURI(URL url) throws URISyntaxException {
+
+        try {
+            return url.toURI();
+        } catch (URISyntaxException e) {
+            try {
+                return new URL(url.getProtocol(),
```

Comment:
Because URLEncoder replaces all special characters he knows about. This includes the '/' characters which makes the url resolution fail.